drivers/FlashIAP.cpp

```diff
@@ -0,0 +1,129 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "FlashIAP.h"
+#include "mbed_assert.h"
+
+#ifdef DEVICE_FLASH
+
+namespace mbed {
+
+FlashIAP::FlashIAP()
+{
+
+}
+
+FlashIAP::~FlashIAP()
+{
+
+}
+
+int FlashIAP::init()
+{
+    if (flash_init(&_flash)) {
+        return -1; 
+    } else {
+        return 0;
+    }
+}
+
+int FlashIAP::deinit()
+{
+    if (flash_free(&_flash)) {
+        return -1; 
+    } else {
+        return 0;
+    }
+}
+
+
+int FlashIAP::read(void *buffer, uint32_t addr, uint32_t size)
+{
```

Comment:
I don't know, in the documentation, it is indicated that `addr` is the address of a block and the function `read blocks` from a flash block device. 
If the operation is done at a byte level maybe you could update the documentation to indicate that `addr` is an an address in the flash block device instead of an address of a block. You could also precise that it reads bytes instead of blocks.